File: apps/web/src/shared/api/auth.api.ts
import { api, clearSession } from './client'

type LoginDto  = { email: string; password: string }
type AuthResult = {
  accessToken: string
  refreshToken: string
  user: { id: string; name: string; email: string; timezone: string; createdAt: string }
}

export const authApi = {
  login:    (dto: LoginDto) => api.post<AuthResult>('/auth/login', dto),
  register: (dto: LoginDto & { name: string; timezone?: string }) => api.post<AuthResult>('/auth/register', dto),
  me:       () => api.get<AuthResult['user']>('/auth/me'),
  refresh:  (refreshToken: string) => api.post<{ accessToken: string; refreshToken: string }>('/auth/refresh', { refreshToken }),
  updateProfile: (patch: { name?: string; email?: string }) =>
    api.patch<AuthResult['user']>('/auth/profile', patch),
  logout:   async () => {
    const refreshToken = typeof localStorage !== 'undefined' ? localStorage.getItem('refresh_token') : null
    if (refreshToken) {
      await api.post('/auth/logout', { refreshToken }).catch(() => {})
    }
    clearSession()
  },
}

File: apps/mobile/lib/shared/api/auth_api.dart
import '../../entities/user/user.dart';
import 'api_client.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

class AuthApi {
  final ApiClient _client;

  AuthApi(this._client);

  /// Returns (user, accessToken, refreshToken)
  Future<(User, String, String)> login(String email, String password) async {
    final res = await _client.post('/auth/login', data: {
      'email': email,
      'password': password,
    });
    final user = User.fromJson(res.data['user'] as Map<String, dynamic>);
    final accessToken = res.data['accessToken'] as String;
    final refreshToken = res.data['refreshToken'] as String;
    return (user, accessToken, refreshToken);
  }

  Future<(User, String, String)> register(String name, String email, String password) async {
    final res = await _client.post('/auth/register', data: {
      'name': name,
      'email': email,
      'password': password,
    });
    final user = User.fromJson(res.data['user'] as Map<String, dynamic>);
    final accessToken = res.data['accessToken'] as String;
    final refreshToken = res.data['refreshToken'] as String;
    return (user, accessToken, refreshToken);
  }

  Future<User> me() async {
    final res = await _client.get('/auth/me');
    return User.fromJson(res.data);
  }

  Future<User> updateProfile({String? name, String? email}) async {
    final res = await _client.patch('/auth/profile', data: {
      if (name != null) 'name': name,
      if (email != null) 'email': email,
    });
    return User.fromJson(res.data);
  }
}

final authApiProvider = Provider((ref) => AuthApi(ref.watch(apiClientProvider)));
